src/Eseye.php

```diff
@@ -323,53 +323,15 @@ public function invoke(string $method, string $endpoint, array $uri_data = []):
         }
 
         // Check if there is a cached response we can return
-        if (in_array(strtolower($method), $this->cachable_verb) &&
-            $cached = $this->getCache()->get($uri)
-        ) {
-            // In case the cached entry is still valid, mark content as being loaded from cache.
-            if (! $cached->expired())
-                $cached->setIsCachedLoad();
-
-            // Handling ETag marked response specifically (ignoring the expired time)
-            // Sending a request with the stored ETag in header - if we have a 304 response, data has not been altered.
-            if ($cached->hasHeader('ETag') && $cached->expired()) {
-
-                $result = $this->rawFetch($method, $uri, $this->getBody(), ['If-None-Match' => $cached->getHeader('ETag')]);
-
-                if ($result->getErrorCode() == 304) {
-
-                    // update expires header with newly provided value
-                    $cached->setExpires($result->expires());
-
-                    // store updated response in cache to renew internal cache duration
-                    $this->getCache()->set($uri, $cached);
-
-                    $cached->setIsCachedLoad();
-                }
-            }
-
-            // In case the result is effectively retrieved from cache,
-            // return the cached element.
-            if ($cached->isCachedLoad()) {
-
-                // Perform some debug logging
-                $this->getLogger()->debug('Loaded cached response for ' . $method . ' -> ' . $uri, [
-                    'etag' => $cached->hasHeader('ETag') ? $cached->getHeader('ETag') : '',
-                    'body' => $cached->raw,
-                ]);
-
-                $this->cleanupRequestData();
-
-                return $cached;
-            }
-        }
+        if ($this->isCachable($method))
+            $result = $this->handleCacheEntry($method, $uri);
 
         // Call ESI itself and get the EsiResponse in case it has not already been handled with cache control
         if (! isset($result))
             $result = $this->rawFetch($method, $uri, $this->getBody());
 
         // Cache the response if it was a get and is not already expired
-        if (in_array(strtolower($method), $this->cachable_verb) && ! $result->expired())
+        if ($this->isCachable($method) && ! $result->expired())
```

Comment:
instead checking if only the method itself can be cache, apply the overall check on the HTTP response.

```suggestion
        if ($this->isCachable($result))
```